dashboard/src/features/projects/common/hooks/useServiceStatus/useServiceStatus.ts

```diff
@@ -0,0 +1,100 @@
+import { useCurrentWorkspaceAndProject } from '@/features/projects/common/hooks/useCurrentWorkspaceAndProject';
+import { useIsPlatform } from '@/features/projects/common/hooks/useIsPlatform';
+import type { ServiceHealthInfo } from '@/features/projects/overview/health';
+import {
+  ServiceState,
+  useGetProjectServicesHealthQuery,
+  type GetProjectServicesHealthQuery,
+  type GetProjectServicesHealthQueryVariables,
+} from '@/utils/__generated__/graphql';
+import type { QueryHookOptions } from '@apollo/client';
+import { useVisibilityChange } from '@uidotdev/usehooks';
+import { useEffect } from 'react';
+
+export interface UseServiceStatusOptions
+  extends QueryHookOptions<
+    GetProjectServicesHealthQuery,
+    GetProjectServicesHealthQueryVariables
+  > {
+  shouldPoll?: boolean;
+}
+
+export default function useServiceStatus(
+  options: UseServiceStatusOptions = {},
+) {
+  const isPlatform = useIsPlatform();
+  const { currentProject } = useCurrentWorkspaceAndProject();
+
+  const isVisible = useVisibilityChange();
+
+  const { data, loading, refetch, startPolling, stopPolling } =
+    useGetProjectServicesHealthQuery({
+      ...options,
+      variables: { ...options.variables, appId: currentProject?.id },
+      skip: !isPlatform || !currentProject,
+      pollInterval: 0,
+      skipPollAttempt: () => !isVisible,
+    });
+
+  // Fetch when mounted
+  useEffect(() => {
+    refetch();
+    return () => {
+      stopPolling();
```

Comment:
I think this is redundant here, see Line 77

Reply:
Not sure, if there's no Line 77, then useEffect wouldn't stop polling after cleanup (when `options.shouldPoll=true`), right?

Maybe it's clearer to rewrite as:
```
useEffect(() => {
    if (options.shouldPoll) {
      if (
        [auth, storage, postgres, hasura, ai, ...Object.values(run)].some(
          (service) => service?.state !== ServiceState.Running,
        )
      ) {
        // Poll more frequently if a service is not running
        startPolling(options.pollInterval || 10000);
      } else {
        // Poll less frequently when all services are running
        startPolling(60000);
      }
    }
    

    return () => stopPolling();
  }
  ```

Reply:
in that case the cleanup will be done on L43 right — That will be executed when the component unmounts anyway

Reply:
Oh I see your point now, I'll update it :+1: